### ISSUE TITLE
[Bug] Random android crash: Cannot access a disposed object. Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.

### ISSUE THREAD

**Issue by holecekp** [xamarin/Xamarin.Forms#12337]
### Description
I am using AppCenter for reporting crashes. Today I have found the following report:

**System.ObjectDisposedException: Cannot access a disposed object. Object name: 'Xamarin.Forms.Platform.Android.Platform+DefaultRenderer'.**

Stack trace:
![android crash 2 10 2020](https://user-images.githubusercontent.com/24872113/94898255-87cc1500-0491-11eb-9026-49ece21389b7.png)

Android version: 10

Maybe, this could be related to the bug https://github.com/xamarin/Xamarin.Forms/issues/11704, which is also ObjectDisposedException. But the stack traces are different.

### Steps to Reproduce

1. This is report from AppCenter so I have no way to check what happened before the crash. I am testing the application thoroughly so the crash probably occurs randomly.

### Expected Behavior
No crashes.

### Actual Behavior
A crash is reported.

### Basic Information

- Version with issue: Xamarin.Forms 4.8 SR 1
- Last known good version: ?
- IDE: VS 2019
- Platform Target Frameworks:
  - Android: 10
- Nuget Packages:
Microsoft.AppCenter.Crashes (3.4.3)
Xamarin.AndroidX.AppCompat.Resources (1.1.0.1)
Xamarin.AndroidX.MediaRouter (1.1.0.5)
Xamarin.Essentials (1.5.3.2)
Xamarin.Forms (4.8.0.1364)

**Comment by Xarpus** [comment]
This error occurred to me while changing the theme on the Samsung SM-T290.

**Comment by hartez** [comment]
@holecekp Do you have any custom renderers in your application?

**Comment by holecekp** [comment]
@hartez Yes, I am using custom renderers for the following controls:
- **ImageButton** and **Button** - adds a tool tip.
- **Entry**, **Editor**, **SearchBar** - adds ability to hide the software keyboard. Moreover, my custom renderers for Editor and SearchBar expose also CursorPosition and SelectionLength that is in the Entry provided with Xamarin.Forms, but it is missing in the other two controls. The renderer for SearchBar contains also a workaround for a wrong icon color in the dark mode (workaround from https://github.com/xamarin/Xamarin.Forms/issues/9266).
- I have another type of **button** with a custom renderer, which sets the gravity to align the text in the button to the left.

Except for the SearchBar custom renderer, which fixes a color of its icon, none of my custom renders manipulate with colors.

**Comment by jsuarezruiz** [comment]
@holecekp Would you have more related logs in App Center? The error seems to be when updating the background color of a disposed view.

Do you know if the error occurs when changing the theme?. Have the App in the background, change the theme and return to the App.

**Comment by holecekp** [comment]
@jsuarezruiz I have only one report in AppCenter for this particular crash. The problem is that only small portion of crashes is reported in AppCenter. The report is not sent immediately, but it is sent on the next time app start. So if the user uninstalls the app immediately, the report is not sent to the AppCenter.

I am using AppThemeBinding for nearly all colors including background colors. My application indeed crashes under certain condition when the theme is changed in the Android settings. I have reported it some time ago as #11704 (I have been able to reproduce it and create a repro project). The crash has the same exception type (ObjectDisposedException), but it occurs in ButtonRenderer and the stack trace is also different. But it is possible that both issues are related. Maybe, they can have the same cause (AppThemeBinding bug?) and they differ in the control where the crash happens.
